Use GitHub Actions to promote configs

jlebon opened the pull request:
This is the first step in simplifying the release process. Allow users
to simply add the `ok-to-promote` label on the issue which in turn
triggers an action to run the promotion script and open a PR.

jlebon commented:
This is one of the first improvements @lucab and I have come up with. It still requires adding the @coreosbot token as a secret in this repo, so it won't work as is.

You can see what this looks like in my personal forks:
- Issue: https://github.com/jlebon/fedora-coreos-streams/issues/1
- PR created: https://github.com/jlebon/fedora-coreos-config/pull/7

lucab commented:
@bgilbert @dustymabe @miabbott can any of you enable Actions on this repository?

bgilbert commented:
@lucab Done.

jlebon commented:
Updated this now! I tested various ways to try to minimize the access to which the secret has access. In the end, I found the best path is to just create a separate bot account. Then, it doesn't really have any special access at all: the GitHub Action just pushes to its fork, and then opens a PR from its fork to the canonical repo.

I created the bot account (@coreosbot-releng) and added its token as a secret to this repo. However, the account was immediately flagged upon creation (which is why you get a 404 when you visit its page). I sent an email to GitHub support to have it unflagged. So for now, let's hold this.

jlebon commented on .github/workflows/promote-config.yml:
```diff
@@ -0,0 +1,58 @@
+    steps:
+      - name: Extract stream name
+        run: |
+          # XXX: consider using separate labels per stream in the future
```
Yeah, I meant more the latter. Parsing the issue title feels a little odd, but I agree it's still better overall to have a single source of truth.

jlebon commented on .github/workflows/promote-config.yml:
```diff
@@ -0,0 +1,58 @@
+          git config user.name 'CoreOS Bot'
+          git config user.email coreosbot@fedoraproject.org
+          ../fcos-releng-auto/scripts/promote-config.sh ${src_stream}
+      - name: Open pull request
```
Done!

jlebon commented on .github/workflows/promote-config.yml:
```diff
@@ -0,0 +1,58 @@
+        with:
+          token: ${{ secrets.COREOSBOT_REPO_SCOPE }}
+          path: fcos
+          branch: ${{ env.target_stream }}-promotion
```
Meh, I think re-using the same branch name is OK. The action knows to overwrite it each time, and it means the fork repo isn't collecting old branches. It's also a way to ensure that there is only a single promotion PR opened at a time.

jlebon commented:
OK, bot account has been created, so we should be good to merge this now!

miabbott reviewed (approved):
Very cool!  I'll let someone with more direct experience do the merging
